src/frontend/src/optimizer/plan_node/batch_postgres_query.rs

```diff
@@ -0,0 +1,96 @@
+// Copyright 2024 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use pretty_xmlish::XmlNode;
+use risingwave_pb::batch_plan::plan_node::NodeBody;
+use risingwave_pb::batch_plan::PostgresQueryNode;
+
+use super::batch::prelude::*;
+use super::utils::{childless_record, column_names_pretty, Distill};
+use super::{
+    generic, ExprRewritable, PlanBase, PlanRef, ToBatchPb, ToDistributedBatch, ToLocalBatch,
+};
+use crate::error::Result;
+use crate::optimizer::plan_node::expr_visitable::ExprVisitable;
+use crate::optimizer::property::{Distribution, Order};
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct BatchPostgresQuery {
+    pub base: PlanBase<Batch>,
+    pub core: generic::PostgresQuery,
+}
+
+impl BatchPostgresQuery {
+    pub fn new(core: generic::PostgresQuery) -> Self {
+        let base = PlanBase::new_batch_with_core(&core, Distribution::Single, Order::any());
+
+        Self { base, core }
+    }
+
+    pub fn column_names(&self) -> Vec<&str> {
+        self.schema().names_str()
+    }
+
+    pub fn clone_with_dist(&self) -> Self {
+        let base = self.base.clone_with_new_distribution(Distribution::Single);
```

Comment:
I noticed `file_scan` uses `Distribution::SomeShard` instead here. But that's a problem because when using that, 4 instances of the `PostgresQueryExecutor` were spawned. It should only spawn a single one.

I changed it to `Distribution::Single` for `batch_postgres_query.rs` and it fixed it for me.

Just flagging it, in case a similar bug can occur for `file_scan` source. cc @chenzl25